bin/s2_incremental

```diff
@@ -69,15 +113,43 @@ def main(level1_root, s2_aoi, start_date, end_date, pkgdir, output_list):
                     logging.info('granule %s already processed', granule)
                 else:
                     # yes it does
-                    logging.info('level1 dataset %s needs to be procesed', level1)
-                    print(level1, file=out)
+                    logging.info('level1 dataset %s needs to be processed', level1)
+                    print(level1, file=out_stream)
+                    # a more realistic count increment would be len(container.granules)
+                    count += 1
                     break
 
-    if output_list is None:
-        worker(sys.stdout)
-    else:
-        with open(output_list, 'w') as out:
-            worker(out)
+        return count
+
+    def nodes_needed():
+        hours_per_granule = 1.5
+        hours, minutes, _ = [int(x) for x in walltime.split(':')]
+        if minutes > 0:
+            hours += 1
```

Comment:
we can probably leave this check off (if minutes > 0 hours += 1), makes the calculation less conservative.